### ISSUE TITLE
gradient based stopping criterion

### ISSUE THREAD

**Issue by mithodin** [stevengj/nlopt#121]
For gradient based optimization algorithms, a stopping criterion based on the gradient would be useful. One could, for example, check that the norm of the gradient is equal to zero up to some user-specified tolerance.

**Comment by stevengj** [comment]
Yes, this is certainly possible (with modifications to handle active constraints).  One annoyance is that the norm of the gradient is dimensionful, and that makes it especially difficult for users to come up with a reasonable tolerance.

**Comment by helske** [comment]
Has there been any plans to implement this?

**Comment by Benoit-F-Q** [comment]
I agree with helske, but the gradient tolerance `tolg` is not settable by the user.

In our team, we patched the hardcoded value.
I wish to share something more useable (open a PR), but I wonder where to add a "luksan_tolg" setter which would be meaningless for a majority of algorithms
